internal/daemonset/daemonset.go

```diff
@@ -0,0 +1,174 @@
+				DNSPolicy:          corev1.DNSClusterFirstWithHostNet,
+				Containers: []corev1.Container{
+					{
+						Name:            "nfd-topology-updater",
+						Image:           nfdInstance.Spec.Operand.ImagePath(),
+						ImagePullPolicy: getImagePullPolicy(nfdInstance),
+						Command: []string{
+							"nfd-topology-updater",
+						},
+						Args:            getArgs(nfdInstance),
+						Env:             getEnvs(),
+						SecurityContext: getSecurityContext(),
+						VolumeMounts:    getVolumeMounts(),
+					},
+				},
+				Volumes: getVolumes(),
+			},
+		},
+	}
+	return controllerutil.SetControllerReference(nfdInstance, topologyDS, d.scheme)
+}
+
+func getImagePullPolicy(nfdInstance *nfdv1.NodeFeatureDiscovery) corev1.PullPolicy {
+	if nfdInstance.Spec.Operand.ImagePullPolicy != "" {
+		return corev1.PullPolicy(nfdInstance.Spec.Operand.ImagePullPolicy)
+	}
+	return corev1.PullAlways
+}
+
+func getArgs(nfdInstance *nfdv1.NodeFeatureDiscovery) []string {
+	return []string{
+		"-kubelet-config-uri=file:/host-var/lib/kubelet/config.yaml",
+		"-podresources-socket=/host-var/lib/kubelet/pod-resources/kubelet.sock",
+		"-sleep-interval=3s",
+	}
+}
+
+func getEnvs() []corev1.EnvVar {
+	return []corev1.EnvVar{
+		{
+			Name: "NODE_NAME",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					FieldPath: "spec.nodeName",
+				},
+			},
+		},
+	}
+}
+
+func getSecurityContext() *corev1.SecurityContext {
+	return &corev1.SecurityContext{
+		RunAsUser: ptr.To[int64](0),
+		SELinuxOptions: &corev1.SELinuxOptions{
+			Type: "container_runtime_t",
+		},
+		ReadOnlyRootFilesystem: ptr.To(true),
+		Capabilities: &corev1.Capabilities{
+			Drop: []corev1.Capability{"ALL"},
+		},
+		AllowPrivilegeEscalation: ptr.To(true),
+	}
+}
+
+func getVolumeMounts() []corev1.VolumeMount {
+	return []corev1.VolumeMount{
+		{
+			Name:      "kubelet-podresources-conf",
+			MountPath: "/host-var/lib/kubelet/config.yaml",
+			ReadOnly:  true,
+		},
+		{
+			Name:      "kubelet-podresources-sock",
+			MountPath: "/host-var/lib/kubelet/pod-resources/kubelet.sock",
+		},
+		{
+			Name:      "host-sys",
+			MountPath: "/host-sys",
+		},
+	}
+}
+
+func getVolumes() []corev1.Volume {
+	return []corev1.Volume{
+		{
+			Name: "kubelet-podresources-conf",
+			VolumeSource: corev1.VolumeSource{
+				HostPath: &corev1.HostPathVolumeSource{
+					Path: "/etc/kubernetes/kubelet.conf",
+					Type: ptr.To[corev1.HostPathType](corev1.HostPathFile),
+				},
+			},
+		},
+		{
+			Name: "kubelet-podresources-sock",
+			VolumeSource: corev1.VolumeSource{
+				HostPath: &corev1.HostPathVolumeSource{
+					Path: "/var/lib/kubelet/pod-resources/kubelet.sock",
+					Type: ptr.To[corev1.HostPathType](corev1.HostPathSocket),
+				},
+			},
+		},
+		{
+			Name: "host-sys",
+			VolumeSource: corev1.VolumeSource{
+				HostPath: &corev1.HostPathVolumeSource{
+					Path: "/sys",
+					Type: ptr.To[corev1.HostPathType](corev1.HostPathDirectory),
+				},
+			},
+		},
```

Comment:
I think we'd want to add the volume (and mount) for `-kubelet-state-dir` i.e. `/var/lib/kubelet` from the host